### PR TITLE
add error checks of devfs_register()

### PR DESCRIPTION
**Pull request by Hxinrong**
### Issues/PRs references

Addresses parts of #15524

### CODE REVIEW

File: boards/mulle/board.c [hunk 1]
```diff
@@ -175,7 +175,10 @@ static int mulle_nvram_init(void)
     }
 
     /* Register DevFS node */
-    devfs_register(&mulle_nvram_devfs);
+    int res = devfs_register(&mulle_nvram_devfs);
+    if (res != 0) {
+        return res;
+    }
 
     return 0;
```
Comment (fjmolinas) [inline boards/mulle/board.c:183]: Why not?
```suggestion
    return devfs_register(&mulle_nvram_devfs);
```

**Comment by fjmolinas** [comment]
Thanks for the fix @Hxinrong! Have you taken a look at our [contributing guidelines](https://github.com/miri64/RIOT/blob/master/CONTRIBUTING.md)? Could you take a look at the [commit convention](https://github.com/miri64/RIOT/blob/master/CONTRIBUTING.md#commit-conventions) :)

You can squash right away once my comment is addressed.

**Comment by fjmolinas** [comment]
@Hxinrong looks good please squash :)

File: boards/mulle/board.c (hunk 1 above)
Comment (Hxinrong) [reply boards/mulle/board.c]: Yes, it will be more tidy.

**Comment by Hxinrong** [comment]
emmm... It seems like I have done something wrong...

**Comment by fjmolinas** [comment]
> emmm... It seems like I have done something wrong... maybe I should make a new branch...

Hmm yes seems like you have mixed up the commit history a bit hahaha. What we would like to have is a single commit with a message like `boards/mulle/board: add error checks on devfs_register`.

I'm not sure what you did here exactly, but there is no need to open a new branch you can do as follows (I made a back up of your branch locally in case something goes wrong :).

```
# reset the branch to match with master, changes will remain but not committed
git reset 8b8a3ffbd93e45ef6366d884ed375e28814891af
# add only the stuff we are interested in this PR
git add boards/
# add a meaningfull commit
git commit -m "boards/mulle/board: add error checks on devfs_register"
# check that the history looks good
git log
# if all looks good push force
git push -f
```

Don't hesitate to ask questions if anything I'm suggestin above is unclear.

**Review by fjmolinas — APPROVED** [review]
ACK.

**Pull request merged**